orchestrator/graphql/resolvers/subscription.py

```diff
@@ -66,24 +78,23 @@ def get_subscription_details(subscription: SubscriptionTable) -> SubscriptionInt
         subscription_details, SubscriptionLifecycle.INITIAL, skip_validation=True
     )
     base_subscription_details.status = subscription_details.status
-    strawberry_type = GRAPHQL_MODELS[graphql_subscription_name(base_model.__name__)]  # type: ignore
-    return strawberry_type.from_pydantic(base_subscription_details)
+    strawberry_type = get_subscription_graphql_type(info, graphql_subscription_name(base_model.__name__))  # type: ignore
+    return strawberry_type.from_pydantic(base_subscription_details)  # type:ignore
 
 
-def format_base_subscription(subscription: SubscriptionTable) -> SubscriptionInterface:
-    from orchestrator.graphql import GRAPHQL_MODELS
+def format_subscription(info: OrchestratorInfo, subscription: SubscriptionTable) -> SubscriptionInterface:
+    if _is_subscription_detailed(info):
+        return get_subscription_details(info, subscription)
 
-    strawberry_type = GRAPHQL_MODELS["subscription"]
-    return strawberry_type.from_pydantic(subscription)
+    strawberry_type = get_subscription_graphql_type(info, "subscription")
```

Comment:
The parameter value "subscription" looks a bit weird / redundant. I guess we also have a class with name Subscription?

Reply:
Ok... I saw some code later and this seems to be the base class. Why not have that very explicit in this code:

```
if _is_subscription_detailed(info):
     return get_subscription_details(info, subscription)
return get_subscription_base(info, subscription)
```

Reply:
Its the base class and also needs to be build from the `subscription_interface` parameter that is given to `create_graphql_router`